data/modules/Assassination/Assassination.lua

```diff
@@ -170,7 +171,16 @@ local makeAdvert = function (station)
 	local time = Engine.rand:Number(1, 4)
 	local due = Game.time + dist / max_ass_dist * time * 22*60*60*24 + Engine.rand:Number(7*60*60*24, 31*60*60*24)
 	local danger = Engine.rand:Integer(1,4)
-	local reward = Engine.rand:Number(2100, 7000) * danger
+
+	local tradeCargo = Game.player.totalCargo*0.81 --the rest 0.19 is used for fuel.
+	local rewardRatio = 3/4-- For getting that ratio of average effective trading profit
+
+	--- Fighters should have independed from cargo reward. For that we increase minimum reward depending on number of missiles that player can acquire.
+	local missiles = Game.player:GetEquipSlotCapacity("missile")
+	local minimumRewardRatio = math.max(rewardRatio, Calibration.Fighting(missiles))
+
+	local profit = math.max(Calibration.profit(tradeCargo) * rewardRatio, Calibration.minProfit * minimumRewardRatio)  
+	local reward = Engine.rand:Number(2100/8259 * profit, 7000/8259 * profit) * danger --8259 is the current estimated profit from assassination module.
```

Comment:
I'll echo laarmen in saying that I don't think this is the right direction; scaling mission rewards just based on how many missiles the player has and how much cargo space they have is logically disconnected from the actual mission module and cheapens the simulation. Instead, better dependent variables to tie to that information would be how many enemies to spawn, or how effective in combat those enemies are, or even what ship and how much equipment the enemy should have. TL;DR: nice idea, but wrong set of variables and it doesn't have the right semantic effect.

Reply:
Those developers that do not use trading when they are  playing, is natural to think that trading should have nothing to do with missions rewards.
Those developers that do not use missions, once they can trade investment of 144000credits, when they are  playing, is natural to think that they should not touch missions rewards.
The big question, that should be asked and eventually answered if possible, is
_**What is the role of missions reward compared to maximum profit from trading?**_
Lets look at the figure of current state that I upload above, on my reply to @laarmen 
Missions are clearly auxiliary to the maximum trading profit that is constant for ever. There is no point to reference on the start of effective trade or on the end of it, because it is constant.
Now let add an exponential  to 'rn' and look at the same figure again, as it appears in figure 9 of my article
![avg_restriction_state](https://user-images.githubusercontent.com/5658995/134545352-76a8245b-4502-4c04-a659-a9d39e930320.png)
We see that, by decreasing the initial trade profit, missions are in considerably better place compared to trading and looses that place gradually, getting to their current place when trading profit gets in its maximum, that by purpose equals to current state's one. 
Now come to my place when i first saw that figure.
**Do developers want missions to be auxiliary, till player starts to effectively trade precious metals?**
If yes, I should  reduce missions' reward to the same ratio that precious metal profit is reduced from maximum trading profit, by adding the exponential.
**Do developers want missions to be auxiliary only against maximum trading profit**?
Then i should not touch missions reward. In that case, _missions would neutralized early and perhaps middle trading._
**Do developers want missions to be auxiliary or even equal to current player's trading profit**, as they used to because, in current state of the game, that trading profit happens to be constant?
In that case, i should normalized missions reward in a similar way that i did here, in assassination module.
But exponential does not know if player is a mercenary ship. The normalization of mission rewards fix all ships rewards, ships that in reality are multi role ships as long they can carry missiles, but reduces to nothing rewards from pure mercenary ships. So, i introduced an artificial linear ratio, that applies to minimum trading profit, that depends on missiles.

Fill free to reject the code, alter it or whatever. If we do nothing with missions but we agree that an exponential should be added, missions will overcome most of trading progress. That means that player will trade only with the largest of ships.

_Missions rewards are estimated by simulating them by wxmaxima code, that can be checked against the existing lua code for them._

Having said that, missions rewards have nothing to do with exponential and the main scope of this PR. Modify or reject the code at will.